src/server/game/map/tile/yield.ts

```diff
@@ -1,16 +1,115 @@
+export type YieldParams = {
+  food?: number,
+  production?: number
+};
+
 export class Yield {
   food: number;
   production: number;
 
-  constructor(params: { food?: number, production?: number }) {
+  constructor(params: YieldParams) {
     this.food = params.food ?? 0;
     this.production = params.production ?? 0;
   }
 
-  add(other: Yield): Yield {
+  copy(): Yield {
+    return new Yield({
+      food: this.food,
+      production: this.production,
+    });
+  }
+
+  incr(other: Yield): Yield {
+    this.food += other.food;
+    this.production += other.production;
+
+    return this;
+  }
+
+  decr(other: Yield): Yield {
+    this.food -= other.food;
+    this.production -= other.production;
+
+    return this;
+  }
+
+  div(other: Yield): number {
+    return Math.max(...[
+      this.food / other.food,
+      this.production / other.production,
+    ].map(num => !isNaN(num) && num ? num : 0));
+  }
+
+  divNumber(n: number): Yield {
     return new Yield({
-      food: this.food + other.food,
-      production: this.production + other.production,
+      food: this.food / n,
+      production: this.production / n,
     });
   }
+
+  add(other: Yield): Yield {
+    return this.copy().incr(other);
+  }
+
+  sub(other: Yield): Yield {
+    return this.copy().decr(other);
+  }
+
+  static max(a: YieldParams, b: YieldParams): YieldParams {
+    return {
+      food: Math.max((a.food ?? 0), (b.food ?? 0)) || undefined,
```

Comment:
Is it intentional that `max(Yield({food: 0}), Yield({food: 0}))` would return `Yield({food: undefined})`?

Reply:
Yes, since the intended input type is `YieldParams` and not `Yield`.

Reply:
The output type is also `YieldParams` instead of `Yield`.